R/responsibilities.R

```diff
@@ -283,14 +283,14 @@ transform_responsibilities <- function(e, t_so_far) {
   # Start with non-changing stuff - NA
   # for burden_sets, and is_open = TRUE
 
-  res$responsibility <- data_frame(
+  res[['responsibility']] <- data_frame(
```

Comment:
It's a paranoid thing more than necessary right here - '$' in R auto-completes, and I have (potentially) `res$responsiblity_set` and `res$responsibility`. If I get the code wrong, nasty things like this can happen...

```
> res <- list()
> res$responsibility_set <- 20
> res$responsibility <- res$responsibility + 5
> res
$responsibility_set
[1] 20

$responsibility
[1] 25
```

whereas...

```
> res <- list()
> res[['responsibility_set']] <- 20
> res[['responsibility']] <- res[['responsibility']] + 5
> res
$responsibility_set
[1] 20

$responsibility
numeric(0)
```

Reply:
So more likely to spot an error if I use [['...']] - the above came out of my newly installed R at home. I should also so
```
options(warnPartialMatchAttr = TRUE)
options(warnPartialMatchDollar = TRUE)
options(warnPartialMatchArgs = TRUE)
```

to get some warnings out when R is auto-completing.

Reply:
Ahhh, I see. Yes partial matching is nasty.

Reply:
This was a bit of a dry-by fix though - I had to do some more work on responsibilities in the fast-forward code, and on reading the responsibilities code felt I should tidy those up. responsibiltiy vs responsibility_set is a bit of a hazard if I get things out of order - made even more hazardous by responsibility_comment and responsibility_set_comment